geometry/intermediate.py

```diff
@@ -0,0 +1,60 @@
+class Coordinate:
+    def __init__(self, x, y):
+        self.x = x
+        self.y = y
+
+
+class LineSegment:
+    def __init__(self, start, end):
+        self.start = start
+        self.end = end
+
+
+def do_line_segments_intersect(line_segment_1, line_segment_2):
+    """
+    Checks to see if two line segments have at least one point that they share
+    """
+    line_segment_2_start_orientation_relative_to_line_segment_1 = calculate_point_triplet_orientation(line_segment_1.start, line_segment_1.end, line_segment_2.start)
+    line_segment_2_end_orientation_relative_to_line_segment_1 = calculate_point_triplet_orientation(line_segment_1.start, line_segment_1.end, line_segment_2.end)
+    line_segment_1_start_orientation_relative_to_line_segment_2 = calculate_point_triplet_orientation(line_segment_2.start, line_segment_2.end, line_segment_1.start)
+    line_segment_1_end_orientation_relative_to_line_segment_2 = calculate_point_triplet_orientation(line_segment_2.start, line_segment_2.end, line_segment_1.end)
+
+    does_line_segment_2_have_one_end_clockwise_to_line_segment_1 = line_segment_2_start_orientation_relative_to_line_segment_1 * line_segment_2_end_orientation_relative_to_line_segment_1 < 0
+    does_line_segment_1_have_one_end_clockwise_to_line_segment_2 = line_segment_1_start_orientation_relative_to_line_segment_2 * line_segment_1_end_orientation_relative_to_line_segment_2 < 0
+
+    if does_line_segment_2_have_one_end_clockwise_to_line_segment_1 and does_line_segment_1_have_one_end_clockwise_to_line_segment_2:
```

Comment:
I bet that if I draw this I would understand, but it's pretty complicated to see, even with the extra long variable names.